conans/client/graph/proxy.py

```diff
@@ -62,67 +68,95 @@ def _get_recipe(self, reference, check_updates, update, remotes):
-            upstream_manifest, ref = self._remote_manager.get_recipe_manifest(ref, selected_remote)
-        except NotFoundException:
-            status = RECIPE_NOT_IN_REMOTE
-            return conanfile_path, status, selected_remote, ref
-
-        read_manifest = recipe_layout.recipe_manifest()
-        # TODO: cache2.0 check this for 2.0
-        if upstream_manifest != read_manifest:
-            if upstream_manifest.time > read_manifest.time:
-                if update:
-                    DiskRemover().remove_recipe(recipe_layout, output=output)
-                    output.info("Retrieving from remote '%s'..." % selected_remote.name)
-                    self._download_recipe(recipe_layout, ref, output, remotes, selected_remote)
-                    status = RECIPE_UPDATED
-                    return conanfile_path, status, selected_remote, ref
+            remote, latest_rrev, remote_time = self._get_rrev_from_remotes(reference,
+                                                                           remotes.values(),
+                                                                           check_all_servers=True)
+            if latest_rrev:
+                # check if we already have the latest in local cache
+                # TODO: cache2.0 here if we already have a revision in the cache but we add the
+                #  --update argument and we find that same revision in server, we will not
+                #  download anything but we will UPDATE the date of that revision in the
+                #  local cache and WE ARE ALSO UPDATING THE REMOTE
+                #  Check if this is the flow we want to follow
+                cache_time = self._cache.get_timestamp(ref)
+                if latest_rrev.revision != ref.revision:
+                    if cache_time < remote_time:
+                        remotes.select(remote.name)
+                        # the remote one is newer
+                        output.info("Retrieving from remote '%s'..." % remote.name)
+                        remote, new_ref = self._download_recipe(latest_rrev, output,
+                                                                remotes, remote)
+                        new_recipe_layout = self._cache.ref_layout(new_ref)
+                        new_conanfile_path = new_recipe_layout.conanfile()
+                        status = RECIPE_UPDATED
+                        return new_conanfile_path, status, remote, new_ref
+                    else:
+                        status = RECIPE_NEWER
                 else:
-                    status = RECIPE_UPDATEABLE
+                    # TODO: cache2.0 we are returning RECIPE_UPDATED just because we are updating
+                    #  the date
+                    if cache_time >= remote_time:
+                        status = RECIPE_INCACHE
+                    else:
+                        selected_remote = remote
+                        remotes.select(remote.name)
```

Comment:
This ``remotes.select()`` here seems very weird. I think the ``select()`` with only 1 remote as possible selection needs to dissapear. But also, this wouldn't be a place to define a "selection" of remotes, that is intended for user input, isn't it?